include/picongpu/fields/incidentField/profiles/SpectralLaser.hpp

```diff
@@ -0,0 +1,309 @@
+                                    + this->getCircularPolarizationVector2() * getValue(totalCellIdx, 0.0_X);
+                            }
+                        }
+
+                        /** The following two functions provide the electric field in frequency domain
+                        * E(Omega) = E_amp * exp(-i*phi)
+                        *
+                        * @param totalCellIdx cell index in the total domain (including all moving window slides)
+                        * @param Omega frequency for which the E-value is wanted
+                        */
+                        HDINLINE float_X E_amp(floatD_X const& totalCellIdx, float_X const Omega) const
+                        {
+                            // transform to 3d internal coordinate system
+                            float3_X pos = this->getInternalCoordinates(totalCellIdx);
+
+                            // calculate focus position relative to the current point in the propagation direction
+                            auto const focusRelativeToOrigin = float3_X(
+                                                                   Unitless::FOCUS_POSITION_X,
+                                                                   Unitless::FOCUS_POSITION_Y,
+                                                                   Unitless::FOCUS_POSITION_Z)
+																   - this->origin;
+                            // current distance to focus position
+                            float_X const focusPos = math::sqrt(pmacc::math::abs2(focusRelativeToOrigin)) - pos[0];
+                            // beam waist at the generation plane so that at focus we will get W0
+                            float_X const waist = Unitless::W0
+                                * math::sqrt(1.0_X + (focusPos / Unitless::R) * (focusPos / Unitless::R));
+
+                            auto planeNoNormal = float3_X::create(1.0_X);
+                            planeNoNormal[0] = 0.0_X;
+                            auto const transversalDistanceSquared = pmacc::math::abs2(pos * planeNoNormal);
+
+                            auto const r2OverW2 = transversalDistanceSquared / waist / waist;
+
+                            // central frequency Unitless::w
+                            // gaussian envelope in frequency domain
+                            // using pulse length - bandwidth product: sigma_Omega = sqrt(2)/tau
+                            float_X norm = math::sqrt(pmacc::math::Pi<float_X>::halfValue * 2.0_X) * Unitless::PULSE_LENGTH * 2.0_X;
+                            float_X eps = norm * math::exp(-(Omega - Unitless::w) * (Omega - Unitless::w) * Unitless::PULSE_LENGTH * Unitless::PULSE_LENGTH);
+                            
+                            float_X amp = eps * math::exp(-r2OverW2);
+
+                            // distinguish between dimensions
+                            if(simDim == DIM2)

```

Comment:
should be `if constexpr(simDim == DIM2)` this will enforce the compile to evaluate this part at compile time.

Reply:
Done